tests/network/libs/bgp.py

```diff
@@ -0,0 +1,247 @@
+import json
+from collections.abc import Generator
+from contextlib import contextmanager
+from typing import Final
+
+import ocp_resources.network_config_openshift_io as openshift_nc
+from kubernetes.dynamic import DynamicClient
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+from ocp_resources.bgp_session_state import BGPSessionState
+from ocp_resources.deployment import Deployment
+from ocp_resources.frr_configuration import FRRConfiguration
+from ocp_resources.namespace import Namespace
+from ocp_resources.pod import Pod
+from ocp_resources.resource import ResourceEditor
+from ocp_resources.route_advertisements import RouteAdvertisements
+from timeout_sampler import retry
+
+from utilities.infra import get_resources_by_name_prefix
+
+_BGP_ASN: Final[int] = 64512
+_EXTERNAL_FRR_IMAGE: Final[str] = "quay.io/frrouting/frr:9.1.2"
+_FRR_DEPLOYMENT_NAME: Final[str] = "frr-k8s-webhook-server"
+_FRR_NS_NAME: Final[str] = "openshift-frr-k8s"
+
+
+@contextmanager
+def enable_route_advertisements_in_cluster(
+    network_resource: openshift_nc.Network, client: DynamicClient
+) -> Generator[None]:
+    """Enables route advertisements in the cluster network resource and deploys the FRR deployment.
+
+    After the context is exited, the changes are reverted and the FRR namespace is cleaned up.
+    The cleanup is expected by the un-patching of the changes (ResourceEditor cleanup).
+    However, it has been observed that the NS is not removed
+    and therefore an explicit delete on the NS is performed.
+
+    Args:
+        network_resource (openshift_nc.Network): The cluster network resource to be patched.
+        client: DynamicClient: The Kubernetes dynamic client.
+
+    Yields:
+        None
+    """
+    patch = {
+        network_resource: {
+            "spec": {
+                "additionalRoutingCapabilities": {"providers": ["FRR"]},
+                "defaultNetwork": {"ovnKubernetesConfig": {"routeAdvertisements": "Enabled"}},
+            }
+        }
+    }
+
+    with ResourceEditor(patches=patch):
+        deployment = Deployment(name=_FRR_DEPLOYMENT_NAME, namespace=_FRR_NS_NAME, client=client)
+        deployment.wait_for_replicas()
+
+        yield
+
+    Namespace(name=_FRR_NS_NAME, client=client).clean_up()
```

Comment:
I don't understand why in `enable_route_advertisements_in_cluster` function you delete ns which is not created by this function